src/main/java/service/ReporterImpl.java

```diff
@@ -0,0 +1,39 @@
+package service;
+
+import dao.Reader;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import validator.Validator;
+
+public class ReporterImpl implements Reporter {
+    private Validator validator;
+    private Reader reader;
+    private OperationStrategy operationStrategy;
+
+    public ReporterImpl(Validator validator, Reader reader, OperationStrategy operationStrategy) {
+        this.validator = validator;
+        this.reader = reader;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public Map<String, Integer> report() {
```

Comment:
```suggestion
    public Map<String, Integer> report(List<String> data) {
```
same here, let's improve flexibility